### ISSUE TITLE
fix build warnings

### ISSUE THREAD

**Issue by lovettchris** [leanprover/elan#46]
there's a ton of them... like this:

```
warning: use of deprecated macro `try`: use the `?` operator instead
    --> src/elan-cli\self_update.rs:1172:22
     |
1172 |     let setup_path = try!(prepare_update());
     |                      ^^^
```

**Comment by NicholasLYang** [comment]
Should be fixed by #53 and #54

**Comment by lovettchris** [comment]
Thanks!
